### PR TITLE
Update evm to support london hardforks

### PR DESCRIPTION
**Pull request by zjb0807**
#1693 

- [x] Update bodhi.js https://github.com/AcalaNetwork/bodhi.js/pull/243
~~- [ ] make access_list to be bounded~~

### CODE REVIEW

File: primitives/src/unchecked_extrinsic.rs [hunk 1]
```diff
@@ -329,6 +329,8 @@ fn verify_eip712_signature(eth_msg: EthereumTransactionMessage, sig: [u8; 65]) -
 	tx_msg.extend_from_slice(&to_bytes(eth_msg.gas_limit));
 	tx_msg.extend_from_slice(&to_bytes(eth_msg.storage_limit));
 	tx_msg.extend_from_slice(&to_bytes(eth_msg.valid_until));
+	// TODO: unimplement
+	//tx_msg.extend_from_slice(&keccak_256(&to_bytes(eth_msg.access_list.as_slice())));
```
Comment (zjb0807) [inline primitives/src/unchecked_extrinsic.rs:333]: TODO

**Comment by ermalkaleci** [comment]
access_list needs to be bounded and I am thinking if it's possible to abuse this feature and make it more expensive to check if read than read itself

**Comment by zjb0807** [comment]
> access_list needs to be bounded and I am thinking if it's possible to abuse this feature and make it more expensive to check if read than read itself

https://github.com/ethereum/go-ethereum/blob/master/core/state_transition.go#L151-L154

In ethereum, the resource size is also limited by the gas limit, which I think is fine. If there is a risk, it will be exposed first in ethereum.

**Comment by zjb0807** [comment]
https://github.com/rust-blockchain/evm/pull/88 merged

**Review by ermalkaleci — DISMISSED** [review]

File: modules/evm/src/lib.rs [hunk 2]
```diff
@@ -1192,9 +1203,6 @@ impl<T: Config> Pallet<T> {
 						Codes::<T>::remove(&contract_info.code_hash);
 						*maybe_code_info = None;
 					}
-				} else {
-					// code info removed while still having reference to it?
-					debug_assert!(false);
```
[the remark this answers is not in the document]
Comment (ermalkaleci) [reply modules/evm/src/lib.rs:1197]: will bring it back, I removed while running evm-tests

**Review by xlc — APPROVED** [review]

**Review by xlc — APPROVED** [review]

**Pull request merged**